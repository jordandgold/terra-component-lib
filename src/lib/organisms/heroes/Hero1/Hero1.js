import React, { Component } from "react";
import "./Hero1.scss";
import CTALinkSection from "../../../CTALinkSection/CTALinkSection";
import PropTypes from "prop-types";
import { windowSizes } from "../../../../utils/globalVariables";

class Hero1 extends Component {
  constructor() {
    super();

    this.state = {
      size: "desktop",
      contentHeight: null
    };
  }

  componentDidMount() {
    window.addEventListener("resize", this.checkSize);

    this.checkSize();
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.checkSize);
  }

  checkSize = () => {
    if (window.innerWidth > windowSizes.desktop) {
      this.setState({
        size: "desktop",
        contentHeight: document.querySelector(
          ".ter-hero-one__content-container"
        ).offsetHeight
      });
    } else if (window.innerWidth > windowSizes.tabletLandscape) {
      this.setState({
        size: "tablet",
        contentHeight: document.querySelector(
          ".ter-hero-one__content-container"
        ).offsetHeight
      });
    } else {
      this.setState({
        size: "mobile",
        contentHeight: document.querySelector(
          ".ter-hero-one__content-container"
        ).offsetHeight
      });
    }
  };

  render() {
    const { images, contentSide, title, text, ctas } = this.props;
    const side = {
      right: "-"
    };
    return (
      <section className="ter-hero-one">
        {this.state.size === "mobile" && (
          <img
            className="ter-hero-one__image ter-hero-one__image--mobile"
            src={images.mobile.url}
            alt={images.mobile.altText}
          />
        )}
        {this.state.size === "tablet" && (
          <img
            className="ter-hero-one__image ter-hero-one__image--tablet"
            src={images.tablet.url}
            alt={images.tablet.altText}
          />
        )}
        {this.state.size === "desktop" && (
          <img
            className="ter-hero-one__image ter-hero-one__image--desktop"
            src={images.desktop.url}
            alt={images.desktop.altText}
          />
        )}
        {contentSide === "right" ? (
          <div
            className="ter-hero-one__content-container"
            style={
              this.state.size === "desktop"
                ? {
                    transform: `translate(-48px, -${this.state.contentHeight +
                      48}px)`,
                    float: "right"
                  }
                : {}
            }
          >
            <h2 className="ter-hero-one__title">{title}</h2>
            <p className="ter-hero-one__text">{text}</p>
            <CTALinkSection ctas={ctas} />
          </div>
        ) : (
          <div
            className="ter-hero-one__content-container"
            style={
              this.state.size === "desktop"
                ? {
                    transform: `translate(48px, -${this.state.contentHeight +
                      48}px)`,
                    float: "left"
                  }
                : {}
            }
          >
            <h2 className="ter-hero-one__title">{title}</h2>
            <p className="ter-hero-one__text">{text}</p>
            <CTALinkSection ctas={ctas} />
          </div>
        )}
      </section>
    );
  }
}

export default Hero1;

Hero1.propTypes = {
  contentSide: PropTypes.string,
  images: PropTypes.shape({
    desktop: PropTypes.shape({
      url: PropTypes.string.isRequired,
      altText: PropTypes.string.isRequired
    }).isRequired,
    tablet: PropTypes.shape({
      url: PropTypes.string.isRequired,
      altText: PropTypes.string.isRequired
    }).isRequired,
    mobile: PropTypes.shape({
      url: PropTypes.string.isRequired,
      altText: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string,
  ctas: PropTypes.shape({
    ctaOne: PropTypes.shape({
      link: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    }),
    ctaTwo: PropTypes.shape({
      link: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    }),
    subCTA: PropTypes.shape({
      link: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    })
  })
};
