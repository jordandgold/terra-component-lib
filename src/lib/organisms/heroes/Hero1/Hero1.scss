@import "../../../global-styles/terra.scss";

.ter-hero-one {
  &__image {
    width: 100%;
  }

  &__content-container {
    background: color("white");
    padding: spacing("md") spacing("sm");
    max-width: 85%;
    margin: auto;
    transform: translateY(-100px);
    box-shadow: $card-box-shadow;

    @media (min-width: $breakpoint-xs) {
      padding: spacing("md") spacing("md");
    }

    @media (min-width: $breakpoint-sm) {
      max-width: 700px;
    }

    @media (min-width: $breakpoint-md) {
      margin: 0;
    }
  }

  &__title {
    font-size: font-size("lg");

    @media (min-width: $breakpoint-sm) {
      font-size: font-size("xl");
    }
  }

  &__text {
    font-size: font-size("base");
    font-weight: font-weight("light");

    @media (min-width: $breakpoint-sm) {
      font-size: font-size("md");
    }
  }

  .ter-cta-section {
    @media (min-width: $breakpoint-xs) {
      max-width: 200px;
    }
  }

  .ter-button {
    display: block;

    &--secondary--1 {
      margin-top: spacing("xs");
    }

    @media (min-width: $breakpoint-md) {
      margin-top: spacing("md");
    }
  }

  .ter-cta-section__link {
    display: block;
    text-align: center;
    font-size: font-size("base");
    margin-top: spacing("xxs");
  }
}
