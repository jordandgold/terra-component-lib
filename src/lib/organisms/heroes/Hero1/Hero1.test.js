import React from "react";
import Hero1 from "./Hero1";
import { shallow, mount } from "enzyme";
import { JSDOM } from "jsdom";

describe("Hero1", () => {
  let wrapper;

  beforeEach(() => {
    const mockCTALinks = {
      ctaOne: {
        className: "ter-button--primary--1",
        text: "Button",
        link: "#"
      },
      ctaTwo: {
        className: "ter-button--secondary--1",
        text: "Button",
        link: "#"
      },
      subCTA: {
        text: "Button",
        link: "#"
      }
    };

    const mockImage = {
      desktop: { url: "./1600x1200.png", altText: "placeholder image" },
      tablet: { url: "./800x600.png", altText: "placeholder image" },
      mobile: { url: "1-to-1.png", altText: "placeholder image" }
    };

    const mockQuerySelector = jest.fn(() => {
      return {
        offsetHeight: 600
      };
    });

    const mockDocument = {
      querySelector: mockQuerySelector
    };

    global.window.innerWidth = 400;

    global.document.querySelector = mockQuerySelector;

    wrapper = mount(
      <Hero1
        images={mockImage}
        title="Design is design."
        text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
        ctas={mockCTALinks}
      />
    );

    const dom = new JSDOM(wrapper.html());
    global.document = dom.window.document;
    global.window = dom.window;
    dom.window.document.querySelector = mockQuerySelector;
  });

  it("should match the snapshot left", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot right", () => {
    const mockCTALinks = {
      ctaOne: {
        className: "ter-button--primary--1",
        text: "Button",
        link: "#"
      },
      ctaTwo: {
        className: "ter-button--secondary--1",
        text: "Button",
        link: "#"
      },
      subCTA: {
        text: "Button",
        link: "#"
      }
    };

    const mockImage = {
      desktop: { url: "./1600x1200.png", altText: "placeholder image" },
      tablet: { url: "./800x600.png", altText: "placeholder image" },
      mobile: { url: "1-to-1.png", altText: "placeholder image" }
    };

    const mockQuerySelector = jest.fn(() => {
      return {
        offsetHeight: 600
      };
    });

    const mockDocument = {
      querySelector: mockQuerySelector
    };

    global.window.innerWidth = 400;

    global.document.querySelector = mockQuerySelector;

    wrapper = mount(
      <Hero1
        images={mockImage}
        title="Design is design."
        text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
        ctas={mockCTALinks}
        contentSide="right"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe("componentWillUnmount", () => {
    it("should call removeEventListener", () => {
      const mockRemoveEventListener = jest.fn();
      global.window.removeEventListener = mockRemoveEventListener;

      wrapper.instance().componentWillUnmount();

      expect(mockRemoveEventListener).toHaveBeenCalled();
    });

    it("should call checkSize", () => {
      const spy = jest.spyOn;
    });
  });

  describe("checkSize", () => {
    it("should set state to desktop", () => {
      global.window.innerWidth = 3000;

      wrapper.setState({ size: "not desktop" });

      wrapper.instance().checkSize();
      const expected = { size: "desktop", contentHeight: 600 };

      expect(wrapper.state()).toEqual(expected);
    });

    it("should set state to tablet", () => {
      global.window.innerWidth = 901;

      wrapper.setState({ size: "not desktop" });

      wrapper.instance().checkSize();
      const expected = { size: "tablet", contentHeight: 600 };

      expect(wrapper.state()).toEqual(expected);
    });

    it("should set state to tablet", () => {
      global.window.innerWidth = 301;

      wrapper.setState({ size: "not desktop" });

      wrapper.instance().checkSize();
      const expected = { size: "mobile", contentHeight: 600 };

      expect(wrapper.state()).toEqual(expected);
    });
  });
});
