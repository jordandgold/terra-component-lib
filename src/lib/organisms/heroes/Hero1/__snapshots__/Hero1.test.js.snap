// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero1 should match the snapshot left 1`] = `
<Hero1
  ctas={
    Object {
      "ctaOne": Object {
        "className": "ter-button--primary--1",
        "link": "#",
        "text": "Button",
      },
      "ctaTwo": Object {
        "className": "ter-button--secondary--1",
        "link": "#",
        "text": "Button",
      },
      "subCTA": Object {
        "link": "#",
        "text": "Button",
      },
    }
  }
  images={
    Object {
      "desktop": Object {
        "altText": "placeholder image",
        "url": "./1600x1200.png",
      },
      "mobile": Object {
        "altText": "placeholder image",
        "url": "1-to-1.png",
      },
      "tablet": Object {
        "altText": "placeholder image",
        "url": "./800x600.png",
      },
    }
  }
  text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
  title="Design is design."
>
  <section
    className="ter-hero-one"
  >
    <img
      alt="placeholder image"
      className="ter-hero-one__image ter-hero-one__image--mobile"
      src="1-to-1.png"
    />
    <div
      className="ter-hero-one__content-container"
      style={Object {}}
    >
      <h2
        className="ter-hero-one__title"
      >
        Design is design.
      </h2>
      <p
        className="ter-hero-one__text"
      >
        Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis.
      </p>
      <CTALinkSection
        ctas={
          Object {
            "ctaOne": Object {
              "className": "ter-button--primary--1",
              "link": "#",
              "text": "Button",
            },
            "ctaTwo": Object {
              "className": "ter-button--secondary--1",
              "link": "#",
              "text": "Button",
            },
            "subCTA": Object {
              "link": "#",
              "text": "Button",
            },
          }
        }
      >
        <section
          className="ter-cta-section"
        >
          <ButtonLink
            className="ter-button--primary--1"
            link="#"
            text="Button"
          >
            <a
              className="ter-button ter-button--primary--1"
              href="#"
            >
              Button
            </a>
          </ButtonLink>
          <ButtonLink
            className="ter-button--secondary--1"
            link="#"
            text="Button"
          >
            <a
              className="ter-button ter-button--secondary--1"
              href="#"
            >
              Button
            </a>
          </ButtonLink>
          <a
            className="ter-cta-section__link"
          >
            Button
          </a>
        </section>
      </CTALinkSection>
    </div>
  </section>
</Hero1>
`;

exports[`Hero1 should match the snapshot right 1`] = `
<Hero1
  contentSide="right"
  ctas={
    Object {
      "ctaOne": Object {
        "className": "ter-button--primary--1",
        "link": "#",
        "text": "Button",
      },
      "ctaTwo": Object {
        "className": "ter-button--secondary--1",
        "link": "#",
        "text": "Button",
      },
      "subCTA": Object {
        "link": "#",
        "text": "Button",
      },
    }
  }
  images={
    Object {
      "desktop": Object {
        "altText": "placeholder image",
        "url": "./1600x1200.png",
      },
      "mobile": Object {
        "altText": "placeholder image",
        "url": "1-to-1.png",
      },
      "tablet": Object {
        "altText": "placeholder image",
        "url": "./800x600.png",
      },
    }
  }
  text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
  title="Design is design."
>
  <section
    className="ter-hero-one"
  >
    <img
      alt="placeholder image"
      className="ter-hero-one__image ter-hero-one__image--mobile"
      src="1-to-1.png"
    />
    <div
      className="ter-hero-one__content-container"
      style={Object {}}
    >
      <h2
        className="ter-hero-one__title"
      >
        Design is design.
      </h2>
      <p
        className="ter-hero-one__text"
      >
        Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis.
      </p>
      <CTALinkSection
        ctas={
          Object {
            "ctaOne": Object {
              "className": "ter-button--primary--1",
              "link": "#",
              "text": "Button",
            },
            "ctaTwo": Object {
              "className": "ter-button--secondary--1",
              "link": "#",
              "text": "Button",
            },
            "subCTA": Object {
              "link": "#",
              "text": "Button",
            },
          }
        }
      >
        <section
          className="ter-cta-section"
        >
          <ButtonLink
            className="ter-button--primary--1"
            link="#"
            text="Button"
          >
            <a
              className="ter-button ter-button--primary--1"
              href="#"
            >
              Button
            </a>
          </ButtonLink>
          <ButtonLink
            className="ter-button--secondary--1"
            link="#"
            text="Button"
          >
            <a
              className="ter-button ter-button--secondary--1"
              href="#"
            >
              Button
            </a>
          </ButtonLink>
          <a
            className="ter-cta-section__link"
          >
            Button
          </a>
        </section>
      </CTALinkSection>
    </div>
  </section>
</Hero1>
`;
